src/item.cpp

```diff
@@ -8201,18 +8203,46 @@ cata::optional<tripoint> item::get_cable_target( player *p, const tripoint &pos
     return g->m.getlocal( source );
 }
 
-bool item::process_cable( player *p, const tripoint &pos )
+bool item::process_cable( player *carrier, const tripoint &pos )
 {
-    const cata::optional<tripoint> source = get_cable_target( p, pos );
+    if( carrier == nullptr ) {
+        reset_cable( carrier );
+        return false;
+    }
+    std::string state = get_var( "state" );
+    if( state == "solar_pack_link" || state == "solar_pack" ) {
+        if( !carrier->has_item( *this ) || ( !carrier->is_wearing( "solarpack_on" ) ||
+                                             !carrier->is_wearing( "solarpack_on" ) ) ) {
```

Comment:
Unless I'm reading this wrong, the check is `X || ( Y || Y )`. Is this intended?

Reply:
You're right, I didn't noticed that. It should be quantum backpack in that second case. Will fix when possible.